test/cql-pytest/test_keyspace.py

```diff
@@ -251,6 +251,22 @@ def test_storage_options_describe(cql, scylla_only):
 
         assert "{'type': 'S3', 'bucket': '42', 'endpoint': 'localhost'}" in desc_output
 
+# Reproducer for scylladb#14139
+def test_alter_keyspace_preserves_udt(cql):
+    cql.execute(f"CREATE KEYSPACE ks WITH REPLICATION = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}}")
+    try:
+        cql.execute(f"CREATE TYPE ks.my_type (my_field int)")
+        cql.execute(f"CREATE TABLE ks.tab (p int PRIMARY KEY, m ks.my_type)")
+        # The contents of the below line doesn't matter, only the fact
+        # that it modifies keyspace metadata.
+        cql.execute(f"ALTER KEYSPACE ks WITH DURABLE_WRITES = false")
+        # If the keyspace-modifying statement messed up keyspace
+        # metadata, the below would fail.
+        cql.execute(f"DROP TABLE ks.tab")
```

Comment:
Weird, I thought the bug was that somehow the my_type definition was ruined. You're saying it was even worse, and the table was also ruined?

Reply:
The bug was that all UDT definitions were being forgotten after ALTER KEYSPACE. (But only in memory — the definitions on disk were fine and restarting the node would fix the problem).

This was making the table definition incomplete and various operations on the table were failing as a result. Originally the issue was observed as `DROP TABLE` crashing the entire cluster.

I just added a `DESCRIBE TYPE` to the test to be more explicit about what the test is testing. (But I left the `DROP TABLE` in).